### DISCUSSION TITLE
get_langchain_handler() alwasy start from trace root.

### DISCUSSION THREAD

**Discussion by hdp173** [langfuse/langfuse#2152] [Support]
I'm working with Langfuse low-level sdk to integrate it with LangChain.
I created a trace with trace() function, and created a span inside that trace.
And then, I get a langchain handler of span() to add generations under that span.
````
trace = langfuse.trace(name="test")
span = trace.span(name="test")
langchain_callback = span.get_langchain_handler()

model = ChatOpenAI(callbacks=[langchain_callback])
````
When I run chain with this model, I expected GENERATION under my span, not under trace directly.
But It always appended under trace directly.
How can I make GENERATIONS added under span?
Please help me.
Thank you.

![image](https://github.com/langfuse/langfuse/assets/35817574/97ba67b5-4e8a-47eb-975a-c48dcd5d845a)

**Comment by marcklingen** [comment]
Can you try whether the behavior is the same when passing the callback handler to the call/invoke/stream method?

**Reply by hdp173** [reply]
It's not working for invoke.
And I need to make it work with ChatOpenAI directly 'cause all of my code is implemented in that way.
Is there any solution for this?

**Reply by hdp173** [reply]
Can I get answer about this issue please?

**Reply by marcklingen** [reply]
thanks for reporting this. can you create an issue as a bug report for this. this is definitely not the way this should work and we are happy to investigate in order to fix this. If you were to pass to call/invoke it'd probably just work but we want to also make this use of the callbackhandler work (although it is discouraged in our docs as Langchain passes inconsistent run_ids to callbackhandlers when added to classes).
